### PR TITLE
Doc for running Meshery on GitHub Codespace

### LINKED ISSUE #5904: [CI] Configure Meshery for Codespaces
#### Current Behavior
Later this year, GitHub Codespaces will be made available to the general public. In the meantime, select open source organizations may engage with Codespaces early to both benefit from the tool and offer feedback on its experience and functionality.

#### Desired Behavior
The ability to open Meshery and it's components in Codespaces via a button like this:
image (9)

#### Implementation
The instructions to make Meshery's repos work with Codespaces looks something like this: [customize your Codespace with a development container](https://docs.github.com/en/codespaces/customizing-your-codespace) ([an example](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/.devcontainer/devcontainer.json)).

#### Acceptance Tests
A single click to get Meshery and its components up and running in Codespaces.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by AnnuCode**
**Notes for Reviewers**

This PR fixes #5904 and also does some more corrections for #6874 (in correcting numbering of steps at one more place).

#### Addition of Codespace doc:

Screenshot 2023-01-17 180338

#### Before:
before-change

#### After:
after-change

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 9a3d92ba738dc4c431bf0f3ac7e0f3901ac39e9a at: https://63d5556e06bfdf16caf07f4a--meshery-docs-preview.netlify.app

File: docs/pages/installation/codespaces.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ <div class="clipboardjs">
+ kubectl config view --minify --flatten > config_minikube.yaml
+ </div></div>
+</pre>
+
+<br />
+Meshery should now be connected with your managed Kubernetes instance. Take a look at the [Meshery guides]({{ site.baseurl }}/guides) for advanced usage tips.
+
+<br />
+#### Note: Run Codespaces on locally installed VSCode IDE for a better experience.
```
Comment (alphaX86) [inline docs/pages/installation/codespaces.md:104]: You could include the instructions to access CodeSpaces VM in VS Code here. It's just feeling blank here

File: docs/pages/installation/codespaces.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+**To Setup and run Meshery on Minikube** :
+
+- [Steps](#steps)
+  - [1. Start minikube](#1-start-minikube)
+  - [2. Install Meshery](#2-install-meshery)
+  - [3. Configure Meshery to use minikube](#3-configure-meshery-to-use-minikube)
+  - [Manual Steps](#manual-steps)
+
+## Steps
+
```
Comment (alphaX86) [inline docs/pages/installation/codespaces.md:27]: Before that, one suggestion we can include "Deploy CodeSpaces" button here.... Refer here... https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/setting-up-your-repository/adding-a-codespaces-badge

**Review by alphaX86 — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
@sivanaikk, will you offer review here?

**Comment by leecalcote** [comment]
@AnnuCode do sign your commits. Follow these instructions - https://github.com/meshery/meshery/pull/6952/checks?check_run_id=10949116737

**Comment by l5io** [comment]
🚀 Preview for commit 5a9ac0cff128a4277cccab2e30d172056bfd2075 at: https://63e6a315cc473a122d202fff--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 84fdd8c464d0fcf16c765597a4c18b81b8ad991a at: https://63e6a5bec1305b187e45a590--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 38ad96ba8d109241d10863b5bf447f2a72065004 at: https://63e7d9d517cc775dd2fda0a9--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit c43e84938ae1569029d722e125357c5014c38de4 at: https://63e7dd2413e94f5d053d253b--meshery-docs-preview.netlify.app

File: docs/pages/installation/codespaces.md (hunk 1 above)
Comment (AnnuCode) [reply docs/pages/installation/codespaces.md]: Sure, thanks. I have updated the doc with an inline link describing how one can open Codespace in the locally installed VS Code along with other options.

File: docs/pages/installation/codespaces.md (hunk 2 above)
Comment (AnnuCode) [reply docs/pages/installation/codespaces.md:27]: Hi, thank you for your suggestion. I have updated the docs with a Codespace badge to open the repo with a single click.

codespace_badge

**Comment by Chadha93** [comment]
@AnnuCode  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by leecalcote — APPROVED** [review]
I went ahead and rewrote the guide.

Thanks for submitting this, @AnnuCode 👍

**Review by leecalcote — APPROVED** [review]

**Comment by AnnuCode** [comment]
> I went ahead and rewrote the guide.
> 
> Thanks for submitting this, @AnnuCode 👍

Thanks a lot for the help @leecalcote !!

**Comment by leecalcote** [comment]
Teamwork. 🙌

**Pull request merged**